added command line option to add additional nameservices

planthaber opened the pull request:
Btw. I think the name "rock-webapp" for this executable is deprecated, what do think about "rock-rest-api" or "rock-http-api" ?

planthaber committed:
added possibility to add additional nameservices

doudou commented:
Instead of adding a new parameter, allow the user to use multiple `--host` options.

doudou commented:
and +1 on the name change (but keep the older name around with a deprecation warning please)

planthaber commented:
I also thought about adding multiple --host, but what if "localhost" should not be included?

doudou commented:
> I also thought about adding multiple --host, but what if "localhost" should not be included?

... I don't understand the remark ...

planthaber commented:
The --host option replaces the main nameserver, --add-host adds nameservers.

When no --host option is given, localhost is the default. By specifiing  rock-webapp --host robot, the rest api only connects to the robot, no matter where the api is running. It does not provide access to Tasks on localhost.

By changing the --host behavior as you suggest, localhost whould always be included.

doudou commented:
> By changing the --host behavior as you suggest, localhost whould always be included.

Actually, the inverse was what I had in mind

```
rock-webapp --host remote-host # only connect to host
rock-webapp --host localhost --host remote-host # connect to host and localhost
```

planthaber committed:
changed --host behavior (fist replaces default, additional --host add hosts)

planthaber committed:
renamed binary to rock-rest-api

planthaber commented:
> rock-webapp --host remote-host # only connect to host
> rock-webapp --host localhost --host remote-host # connect to host and localhost

That is better, indeed. Changed it and also renamed the binary, to see the changes:
https://github.com/planthaber/tools-rest_api/commit/24b22defe4a8a05370cec3855aa8f73bc6630437

planthaber commented:
Any further comments?

doudou commented on bin/rock-rest-api:
```diff
@@ -0,0 +1,128 @@
+
+#add the remaining additional name servers
+name_server_hosts.each do |host_address|
+    Orocos::name_service << Orocos::CORBA::NameService.new(host_address)
```
Use `.` instead of `::` to access methods. Accessing methods from modules using `::` is officially deprecated.

planthaber committed:
Use . instead of :: to access methods.

planthaber commented:
applied

doudou commented on bin/rock-webapp:
```diff
@@ -1,114 +1,3 @@
-end
-thin.run(app, Host: thin_host, Port: thin_port)
-
+puts "'rock-webapp' is deprecated, please use 'rock-rest-api'"
```
Either run rock-rest-api or exit(1). Otherwise, people that start the API as a service will not even get an error. I would personally prefer the first one (exec rock-rest-api) to provide a smoother transition

Moreover, send the message to STDERR.

planthaber committed:
call rock-rest-api when rock-webapp is used

planthaber commented:
also applied, I wondered if sleeping the rock-webapp script before calling rock-rest-api whould draw more attention to the change, but I think this can be done as next step before deleting the old script file
